### ISSUE TITLE
Screen Reader: Many Pages: The dates are announced without any relationship to weekdays, month, year

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#74865]
## Action Performed
Prerequisites:
- The user is signed in

1. Using Chrome+JAWS, open the URL - https://new.expensify.com/
2. Navigate using the tab key to the Account button and press Enter to activate it.
3. Navigate using the tab key to the Security button and press Enter to activate it.
4. Navigate to the Date of Birth field and Press Enter to activate the modal.
5. Navigate to the Calendar icon and activate the calendar icon to open the date picker.
6. Navigate through the dates and observe how the screen reader announcement.

Other Occurrences:
1. On Send Invoice - Date

## Expected Result
Each date cell in the calendar should be semantically associated with its corresponding weekday, month, and year. Screen readers should announce dates in a meaningful way, e.g.:
- 'Tuesday, November 10, 2025'

## Actual Result
Individual date elements are announced only by their numeric value (e.g., “10”), without any context of weekday, month, or year.The lack of programmatic relationships between the date, its weekday, and the current month/year prevents screen readers from conveying meaningful context. As a result, users cannot determine whether “10” refers to October 10, November 10, or Tuesday the 10th.

HTML:
<div class="css-g5y9jx r-18u37iz r-tmtnm0"><div aria-label="3" data-tag="pressable" data-selection-scrapper-hidden-element="true" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21 r-9ffhgg r-1awozwy r-13awgt0 r-tmtnm0 r-1777fci r-lrvibr" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx"><div class="css-g5y9jx r-1awozwy r-1ylenci r-1g7fiml r-1777fci r-1udh08x r-199wky7" style=""><div dir="auto" class="css-146c3p1" style="color: rgb(0, 46, 34); font-size: 15px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 400; line-height: 20px;">3</div></div></div><div aria-label="4" data-tag="pressable" data-selection-scrapper-hidden-element="true" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21 r-9ffhgg r-1awozwy r-13awgt0 r-tmtnm0 r-1777fci r-lrvibr" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx"><div class="css-g5y9jx r-1awozwy r-1ylenci r-1g7fiml r-1777fci r-1udh08x r-199wky7" style=""><div dir="auto" class="css-146c3p1" style="color: rgb(0, 46, 34); font-size: 15px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 400; line-height: 20px;">4</div></div></div><div aria-label="5" data-tag="pressable" data-selection-scrapper-hidden-element="true" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21 r-9ffhgg r-1awozwy r-13awgt0 r-tmtnm0 r-1777fci r-lrvibr" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx"><div class="css-g5y9jx r-1awozwy r-1ylenci r-1g7fiml r-1777fci r-1udh08x r-199wky7" style="background-color: rgb(216, 209, 199);"><div dir="auto" class="css-146c3p1" style="color: rgb(0, 46, 34); font-size: 15px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 400; line-height: 20px;">5</div></div></div><div aria-label="6" data-tag="pressable" data-selection-scrapper-hidden-element="true" tabindex="0" class="css-g5y9jx r-1otgn73 r-1tazni7 r-9ffhgg r-1awozwy r-13awgt0 r-tmtnm0 r-1777fci r-lrvibr r-17b9qp5" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx"><div class="css-g5y9jx r-1awozwy r-1ylenci r-1g7fiml r-1777fci r-1udh08x r-199wky7">

## Area issue was found
Settings-Security

## Failed Checkpoint
WCAG 1.3.1 Info and Relationships
https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships

## User Impact
Users with different disabilities employ assistive technology to adapt content according to their specific needs. For example, users who are blind (using a screen reader) benefit when information conveyed through color is also available in text (including text alternatives for images that use color to convey information), and users who are deaf-blind and use braille (text) refreshable displays may be unable to access color-dependent information.

## Suggested Resolution
Either implement the date picker with a table layout semantics or use aria-label to announce each date together with its whole information, e.g.:
- 'Tuesday, November 10, 2025'
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/

## Environment(s)
Windows Windows 11 - Chrome

## Screenshots/Videos
https://github.com/user-attachments/assets/83f23d21-98ec-43ca-87b5-819dddfa2593

<img width="1919" height="973" alt="Image" src="https://github.com/user-attachments/assets/4b11766a-3eba-4097-8e6d-a59970bf4a59" />

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6994747
**Issue reported by:** NA
**Slack conversation:** NA

**Comment by mallenexpensify** [comment]
`Accessibility` issue, will be addressed soon.
